internal/services/datafactory/data_factory_linked_service_sftp_resource.go

```diff
@@ -244,6 +292,10 @@ func resourceDataFactoryLinkedServiceSFTPRead(d *pluginsdk.ResourceData, meta in
 	d.Set("name", resp.Name)
 	d.Set("data_factory_id", dataFactoryId.ID())
 
+	if _, ok := d.GetOk("private_key_content_base64"); !ok {
+		d.Set("private_key_content_base64", nil)
+	}
```

Comment:
The authUpdate acceptance test was failing because when updating from `private_key_content_base64` to `password`, the import step failed because the subsequent plan showed one change: it wanted to change `private_key_content_base64` from `""` (state / Azure) to `null` (configuration). I don't totally understand why `private_key_content_base64` has this problem and other secure fields do not, but these lines I added here fixed the acceptance test.

Reply:
I suspect this is due to the combined CreateUpdate - we've been splitting these whenever we can as there are additional functional and behavioural benefits to having the code bespoke to the operation (e.g. `d.HasChanges()` does not work as expected in the combined functions.)

You might consider a separate PR to split the function (it's worth the extra effort imo to separate the concerns and have confidence in the changes from the testing) and then rebase this PR onto that when it's in?